### PR TITLE
Refactor helpers for storage key formatting

### PR DESCRIPTION
**Pull request by abronan**
Refactor helpers to have more meaningful names and be more flexible with future usage.

I kept the approach with a boolean to omit the first `/` or not. Because we need this for the `getDirectory` function and I'd like to avoid having a private `normalize` and `getDirectory` functions for each store that needs to omit the first `/` in the future.

Signed-off-by: Alexandre Beslic <abronan@docker.com>

### CODE REVIEW

File: pkg/store/helpers.go [hunk 1]
```diff
@@ -12,21 +12,38 @@ func createEndpoints(addrs []string, scheme string) (entries []string) {
...
+// Normalize the key for each store to the form:
+//
+//     /path/to/key
+//
+// If omit is set to 'true', this will omit
+// the first '/' character:
+//
+//     path/to/key/
+//
+func normalize(key string, omit bool) string {
```
Comment (aluzzardi) [inline pkg/store/helpers.go]: I wouldn't put omit in there probably, it's not useful to anyone but consul.

Comment (abronan) [inline pkg/store/helpers.go]: I'm worried that some K/V (only if we consider adding more backends in the future) may forbid using special characters like `.`, `-`, and especially `/` as the first character of a Key. Either way this is a cosmetic change so I don't mind changing it to have a private method in Consul for now :)

Comment (aluzzardi) [inline pkg/store/helpers.go]: We'll do that once we get there :)

So far, no new backends are planned

**Comment by vieux** [comment]
@abronan please rebase :P

**Comment by aluzzardi** [comment]
LGTM

**Pull request merged**